File: backend/app/schemas/linkedin/linkedin_analyzer.py
from pydantic import BaseModel
from typing import List


class LinkedInAnalyzerRequest(BaseModel):
    language: str = "en"
    headline: str
    about: str
    skills: List[str] = []
    projects: List[str] = []
    target_role: str


class LinkedInSignals(BaseModel):
    headline: int
    about: int
    skills: int
    search_visibility: int


class MissingKeyword(BaseModel):
    keyword: str
    reason: str


class SearchVisibilityItem(BaseModel):
    title: str
    rank: str
    description: str


class RecruiterMatchBreakdown(BaseModel):
    target_role_match: int
    keyword_coverage: int
    search_visibility: int
    profile_clarity: int
    missing_proof_points: List[str]


class LinkedInAnalyzerResponse(BaseModel):
    profile_score: int
    signals: LinkedInSignals
    missing_keywords: List[MissingKeyword]
    headline_rewrite: str
    about_rewrite: str
    recruiter_search_visibility: List[SearchVisibilityItem]
    match_breakdown: RecruiterMatchBreakdown
    ai_conclusion: str

File: frontend/src/features/linkedin/types/linkedinAnalyzer.ts
export interface LinkedInAnalyzerRequest {
  language?: string;

  analysis_language?: string;

  headline: string;

  about: string;

  skills: string[];

  projects: string[];

  target_role: string;
}

export interface LinkedInSignals {
  headline: number;

  about: number;

  skills: number;

  search_visibility: number;
}

export interface MissingKeyword {
  keyword: string;

  reason: string;
}

export interface SearchVisibilityItem {
  title: string;

  rank: string;

  description: string;
}

export interface RecruiterMatchBreakdown {
  target_role_match: number;

  keyword_coverage: number;

  search_visibility: number;

  profile_clarity: number;

  missing_proof_points: string[];
}

export interface LinkedInAnalyzerResponse {
  profile_score: number;

  signals: LinkedInSignals;

  missing_keywords: MissingKeyword[];

  headline_rewrite: string;

  about_rewrite: string;

  recruiter_search_visibility: SearchVisibilityItem[];

  match_breakdown: RecruiterMatchBreakdown;

  ai_conclusion: string;
}

export interface LinkedInProfileData {
  id?: number | null;

  user_id?: number;

  language?: string;

  headline: string;

  about: string;

  skills: string[];

  projects: string[];

  target_role: string;

  analysis?: LinkedInAnalyzerResponse | null;

  latest_profile_score?: number | null;

  analyzed_at?: string | null;
}
